microprofile/WFLY-17137-microprofile-jakarta-ee-10.adoc

```diff
@@ -0,0 +1,202 @@
+
+[cols="1,1,1,1,1"]
+|===
+|Spec | MP 5 Version | Current WildFly Version | MP 6 Version | Updated Functionality (y/n)
+
+|Config
+|3.0
+|3.0
+|3.0
+|n
+
+|Fault Tolerance
+|4.0
+|4.0
+|4.0
+|n
+
+|Health
+|4.0
+|4.0
+|4.0
+|n
+
+|JWT
+|2.0
+|2.0
+|2.1
+|y
+
+|+++<s>Metrics</s>+++ +
+Will not include
+|+++<s>4.0</s>+++
+|+++<s>4.0</s>+++
+|+++<s>5.0</s>+++
+|+++<s>y</s>+++
+
+|OpenAPI
+|3.0
+|3.0
+|3.1
+|y (trivial)
+
+|OpenTracing +
+Dropped in MP 6
+|3.0
+|3.0
+|-
+|-
+
+|Rest Client
+|3.0
+|3.0
+|3.0
+|n
+
+|Telemetry +
+New in MP 6
+|-
+|-
+|1.0
+|y
+
+|===
+
+There will be a separate RFEs for:
+
+* Removal of MicroProfile Metrics (https://issues.redhat.com/browse/WFLY-17138[WFLY-17138]). The functional replacement is Micrometer, listed below.
+* Removal of MicroProfile OpenTracing (https://issues.redhat.com/browse/WFLY-17510/[WFLY-17510]). The functional replacement is Micrometer, listed below.
+* Addition of MicroProfile Telemetry (https://issues.redhat.com/browse/WFLY-17156[WFLY-17156])
+* Addition of Micrometer (https://issues.redhat.com/browse/WFLY-14947[WFLY-14947])
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-17137[WFLY-17137] - Provide Support for MicroProfile.Next version that Aligns with Jakarta EE 10
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1861[EAP7-1861] - Provide Support for MicroProfile.Next version that Aligns with Jakarta EE 10
+* https://issues.redhat.com/browse/WFLY-17138[WFLY-17138] - Convert MP Metrics subsystem to Legacy Mode
+* https://issues.redhat.com/browse/WFLY-17156[WFLY-17156] - Implement Support for MicroProfile Telemetry
+* https://issues.redhat.com/browse/WFLY-17510/[WFLY-17510] - Removal of MicroProfile OpenTracing
+* https://issues.redhat.com/browse/WFLY-14947[WFLY-14947] - Addition of Micrometer
+* https://issues.redhat.com/browse/WFLY-17144[WFLY-17144] - Add Micrometer to WildFly
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+WildFly will be updated to pull in the corresponding versions of the Eclipse Microprofile specification jars, as well as the SmallRye jars and dependencies needed to implement each version.
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
```

Comment:
@bstansberry @darranl  I can never remember. Should managed domain be checked or not for MicroProfile/XP stuff?

Reply:
@kabir If history tells us anything, then leave unchecked.

```
microprofile/WFLY-14798-upgrade-reactive-messaging-2.0.adoc:* [ ] Managed domain
microprofile/WFLY-14932_mp_reactive_messaging_kafka_api.adoc:* [ ] Managed domain
microprofile/WFLY-14940_upgrade-microprofile-health-3.1.adoc:* [ ] Managed domain
microprofile/WFLY-14987_reactive_messaging-ssl-context-kafka-connector.adoc:* [ ] Managed domain
microprofile/WFLY-15832-microconfig-config-root-folder.adoc:* [x] Managed domain
```